uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/request-modal-details-toil.controller.js

```diff
@@ -15,6 +15,10 @@ define([
       hasPreviousExpirationDate: null,
       hasExpirationFromAdminSettings: null
     };
+    var originalRequestDates = {
+      from_date: detailsController.request.from_date,
```

Comment:
Instead of saving individual properties. Save the whole object. For example here, https://github.com/civicrm/civihr/blob/PCHR-3281-dropdown-arrows-not-working-leave-popup/uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request.controller.js#L880

So its better to create a similar function here.